### ISSUE TITLE
Parsing from UnixSocket may be unreliable

### ISSUE THREAD

**Issue by bjoernricks** [greenbone/gvm-tools#60]
In https://github.com/greenbone/gvm-tools/blob/master/gmp/gvm_connection.py#L953 

>             # Todo: Why does the sleep helps here? Sometimes it will break
>             # here because the message is missing some bytes at the end.
>             # Same script and with tls or ssh, then it works flawless without
>             # "sleep()"
> time.sleep(0.000001)

it is already indicated that reading from the unix socket doesn't work reliable. Having to use some sleep timer is always an indicator for flaky behavior.

The Problem with the code here is reading from the socket is aborted if the size of the read data is smaller as the buffer size
```python
if len(data) < BUF_SIZE:
  break
```
IMHO this code will break if gvm-tools will read the data faster then gvmd will send it to the socket. It will even become more unreliable if the os/interpreter/framework is using some internal buffering. This already happened to me with a nodejs based implementation.

The solution would be to read the first chunk of data from the socket and extract the opening root xml element e.g. `<get_tasks_response>`. Afterwards the client code must read from the socket until the closing root element arrived e.g. `</get_tasks_response>`. Also don't forget to add a timer for reading to abort if the server didn't send an answer in time.

**Comment by bjoernricks** [comment]
Btw. the proposed solution is used by gsad in https://github.com/greenbone/gvm-libs/blob/master/util/xmlutils.c#L707 already

**Comment by timopollmeier** [comment]
There is one (somewhat obscure) case where simply checking for the closing tag of the root element wouldn't work: Multiple `<commands>` element nested inside each other.
Maybe we should consider no longer allowing this nesting on the GMP side or even removing the `<commands>` command.

**Comment by bjoernricks** [comment]
@timopollmeier only one root element is allowed for xml documents. Therefore we should never respond with something like
```xml
<get_xyz_response>...</get_xyz_response>
<command_a>...</command_a>
```

Looking at gvm-libs imho it wouldn't even be possible to parse this response.

**Comment by bjoernricks** [comment]
Fixed in #71
